models/migration.go

```diff
@@ -36,23 +36,21 @@ func NewMigration(migrationBytes io.ReadCloser, fileName string) (*Migration, er
 		return nil, fmt.Errorf("could not parse file: %s", fileName)
 	}
 
+	buf := new(bytes.Buffer)
```

Comment:
Can you explain the issue? What was the problem in returning a new string every time?

Reply:
If you try to get the query from a `models.Migration`, it was reading from an `io.ReaderCloser`, which makes re-reads empty since the reader was already read. This way, we read and store in the migration to be able to access the migration query anytime.

Reply:
Aha, thanks!

Reply:
Hmm but the backward and forward migration are two separate commands right? Can you point me to the code which is reading the migration twice?

Reply:
Eh, this happens only when you try to run the code with the following snippet:

```Go
	engine, err := morph.New(context.Background(), driver, src)
	if err != nil {
		return err
	}
	defer engine.Close()

	_, err = engine.Apply(1)
	if err != nil {
		return err
	}
	_, err = engine.ApplyDown(1)
	if err != nil {
		return err
	}
	_, err = engine.Apply(1) // <--- Since we read all of the io.Reader, it returns empty for the migration.
	if err != nil {
		return err
	}
```

Reply:
Oh I see, so this was happening when our morph library was being used? I thought this was happening via CLI commands.

Reply:
Yeah this is not a usual scenario but still a valid bug. When using in CLI mode this wouldn't happen as the file will be read independently each time.